### PR TITLE
Menu item styles and interactivity

### PR DESCRIPTION
**Pull request by petemill**
Menu Item enhancements:
- Styling matches figma (hover, selection, focus)
- Pressing enter on the items actually clicks them
- Items can opt out of built-in styles and tabbing / keyboard down arrow and enter press via using a <div> (maybe it's separators or custom interactivity)
- Items can opt out of the menu immediately closing on selection by specifying `data-is-interactive={true}`

### CODE REVIEW

**Review by nullhook — APPROVED** [review]

**Comment by petemill** [comment]
I merged this without codeowner approval due to PTO and to unblock brave-core. Let me know if anyone has extra feedback.

File: src/components/menu/menu.svelte
Comment (fallaciousreasoning) [inline src/components/menu/menu.svelte:89]: Maybe worth extracting these into an array?

`const menuChildTags = ['LEO-OPTION', 'LEO-MENU-ITEM']`

File: src/components/menu/menu.svelte
Comment (fallaciousreasoning) [inline src/components/menu/menu.svelte:199]: :+1:

File: src/components/menu/menu.svelte
Comment (fallaciousreasoning) [inline src/components/menu/menu.svelte:91]: I think it would be ideal if we didn't have to mark items as `data-interactive=true` - it complicates the API somewhat - is it not possible to use some combination of `stopPropagation` and `preventDefault` to achieve the same result?

Comment (petemill) [reply src/components/menu/menu.svelte:91]: We still want to have `item.click()` fire. In other words, we want to be able to hit kbd arrows and then enter to fire the click event. Perhaps since we're piggybacking on element focus then "enter" might cause the element's onclick to fire? I can't remember. Overall this is probably less complicated if we make Components out of child items (since they seem to have particular design patterns) and expose attributes that affect this explicitly. Or some other way to expose options more cleanly instead of a mix of tag name, data-attributes (and class names) - data attributes everywhere?! 🙃

**Pull request merged**
